apps/shipments/views/tracking_data.py

```diff
@@ -34,32 +34,37 @@ class DeviceViewSet(viewsets.ViewSet):
     permission_classes = permissions.AllowAny
     serializer_class = PermissionLinkSerializer
 
-    @action(detail=True, methods=['post'], permission_classes=(permissions.AllowAny,))
-    def tracking(self, request, version, pk):
-        LOG.debug(f'Adding tracking data by device with id: {pk}.')
-        log_metric('transmission.info', tags={'method': 'devices.tracking', 'module': __name__})
-
+    def _validate_payload(self, request, pk):
```

Comment:
Just a bit of positive feedback, I love how you DRY'd this up between Tracking/Telemetry views